### PR TITLE
feat: Add Fave Functionality

### LINKED ISSUE #14: Add Favorite Functionality
- When a user clicks the star button:
  - the data model should be updated such that you can tell which ideas have been starred.
  - the star should toggle between orange and white to show the user whether that idea is favorited or not.
    - orange: favorited
    - white: not favorited
- When an idea is created, it should start not favorited.

Note: Most of the CSS for this may have been completed in https://github.com/ericahagle/ideabox/issues/13

image

### PR DESCRIPTION
**Pull request by ericahagle**
# Description
- When a user clicks the star button:
  - the data model is updated such that you can tell which ideas have been starred.
  - the star will toggle between orange and white to show the user whether that idea is favorited or not.
    - orange: favorited
    - white: not favorited

# Contributors
@ericahagle 

# Checklist
- [x] My PR has an appropriately descriptive and concise title.
- [x] My PR has a clear description of all proposed changes.
- [x] My PR denotes any/all team members who contributed to it.
- [x] My code follows the Turing Style Guides and best practices.
- [x] I ran the code locally and verified that there are no visible errors.
- [x] feat: My code is clean and readable, so my teammates can easily review it.

Screenshot 2023-09-15 at 4 31 39 PM

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -55,8 +57,8 @@ function showIdea() {
   newIdea.className = "new-idea";
   newIdeaTitle.className = "new-idea-title";
   newIdeaBody.className = "new-idea-body";
-  newIdeaFaveFalse.className = "new-idea-fave-btn";
-  newIdeaFaveTrue.className = "new-idea-fave-btn";
+  newIdeaFaveFalse.className = "new-idea-fave-btn-false";
```
Comment (ben-rosner-williamsburg) [inline main.js]: This feels like you're on the right track, but I wonder if classList.toggle might be more useful here.

File: main.js [hunk 2]
```diff
@@ -104,4 +106,30 @@ outputContainer.addEventListener('click', function(event) {
...
+      for (var j = 0; j < currentIdeas.ideas.length; j++) {
+        if (currentIdeas.ideas[j].id.toString() === faveIdeaId) {
+          faveIdeas.push(currentIdeas.ideas[j]);
+          // newIdeaFaveTrue.classList.add("hidden");
+          // newIdeaFaveFalse.classList.remove("hidden");
+          break;
+        }
+      } 
+    }
+  }
```
Comment (ben-rosner-williamsburg) [inline main.js]: I feel like we're overcomplicating this. There's got to be a simpler way to get this to work.

File: main.js [hunk 3]
```diff
@@ -88,20 +86,49 @@ function showIdea() {
...
+
+outputContainer.addEventListener('click', function(event) {
+  if (event.target.classList.contains("new-idea-fave-btn")) {
+    var faveIdeaId = event.target.closest('.new-idea').getAttribute('data-id');
+    for (var i = 0; i < currentIdeas.ideas.length; i++) {
+      if ((currentIdeas.ideas[i].id.toString() === faveIdeaId) && (currentIdeas.ideas[i].isFave === true)) {
+        currentIdeas.ideas[i].isFave = false;
+        for (var j = 0; j < faveIdeas.length; j++) {
+          if (faveIdeas[j].id.toString() === faveIdeaId) {
+            faveIdeas.splice(j, 1);
```
Comment (ben-rosner-williamsburg) [inline main.js:115]: We technically don't need the 1 as the 2nd argument there I think, but that's OK.

**Review by ben-rosner-williamsburg — APPROVED** [review]
Looks fine to me. Approved.

File: main.js (hunk 1 above)
Comment (ericahagle) [reply main.js]: Ended up not needing either! I was able to just use the one class, and then utilize add and remove of attributes to accomplish what I needed. 😄

File: main.js (hunk 3 above)
Comment (ericahagle) [reply main.js:115]: I was basically mirroring what I saw in MDN, specifying that the count is only 1 for the splice, just to be more explicit.

Comment (ericahagle) [reply main.js:115]: We can look at this again when DRYing and cleaning up the code and see if it makes sense to remove it, cool?

File: main.js (hunk 2 above)
Comment (ericahagle) [reply main.js]: Idk if it was simpler, but I got it to work in the end! lol

**Pull request merged**